### PR TITLE
#184 Converted classic pipeline to yaml - Publish Templates 3.0

### PR DESCRIPTION
**Pull request by hirendra-singh**
Converted classic pipeline to YAML https://github.com/robe070/cookbooks/issues/184

**Classic Pipeline**
repository: robe070/cookbooks
Branch - debug/paas
Release pipeline name - Template Release 2.0

**Converted pipeline to yaml**
repository: hirendra-singh/cookbooks
Branch - hiren/aws-template-release-3.0
Name - Publish Templates 3.0
Pipeline Link - https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build?definitionId=65

### CODE REVIEW

**Comment by hirendra-singh** [comment]
@robe070 there is some version txt match error in "AWS PowerShell: Test Image Version" task.
 
Can you please check/suggest?
 
https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2299&view=logs&j=0ec8d5da-3446-585a-8bd6-593d83005233&t=ba4ba2ab-89e7-51b7-f61f-a37e121e4b79&l=25
 
[Pipelines - Run 20241003.1 logs (](https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2299&view=logs&j=0ec8d5da-3446-585a-8bd6-593d83005233&t=6471ef13-4782-5a1e-32bc-1c8a74c7ed6e)[azure.com](http://azure.com/))

**Review by robe070 — CHANGES_REQUESTED** [review]
We need the maximum time before the task is reported as an error. AWS and Azure marketplace can both take a long time to process this step.

You need to use OnTimeout: resume
https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/manual-validation-v0?view=azure-pipelines#:~:text=onTimeout%20%2D%20On%20timeout%0Astring.%20Allowed%20values%3A%20reject%2C%20resume.%20Default%20value%3A%20reject.

**Comment by robe070** [comment]
> We need the maximum time before the task is reported as an error. AWS and Azure marketplace can both take a long time to process this step.
> 
> You need to use OnTimeout: resume https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/manual-validation-v0?view=azure-pipelines#:~:text=onTimeout%20%2D%20On%20timeout%0Astring.%20Allowed%20values%3A%20reject%2C%20resume.%20Default%20value%3A%20reject.

I notice you haven't used this method. Why is that?

**Comment by robe070** [comment]
Why do you need a new set of variables - UpdatedGate?
This is not the way to do it.
You need to create a new version of `SetGateVariable.ps1`
Call it `SetGateVariableTemplateCI.ps1`
And set the Version in that script using TemplateCI

**Comment by hirendra-singh** [comment]
> > We need the maximum time before the task is reported as an error. AWS and Azure marketplace can both take a long time to process this step.
> > You need to use OnTimeout: resume https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/manual-validation-v0?view=azure-pipelines#:~:text=onTimeout%20%2D%20On%20timeout%0Astring.%20Allowed%20values%3A%20reject%2C%20resume.%20Default%20value%3A%20reject.
> 
> I notice you haven't used this method. Why is that?

Hi @robe070,
I have used **ManualValidation@1** task instead of **ManualIntervention@8**. It has resolved the issue. Let me know if I still need to add **OnTimeout: resume** in **ManualValidation1@**.

**Comment by hirendra-singh** [comment]
> Why do you need a new set of variables - UpdatedGate? This is not the way to do it. You need to create a new version of `SetGateVariable.ps1` Call it `SetGateVariableTemplateCI.ps1` And set the Version in that script using TemplateCI

Ok I will check and create new version SetGateVariable.ps1 and will set the TemplateCI version in script.

**Comment by robe070** [comment]
> Hi @robe070, I have used **ManualValidation@1** task instead of **ManualIntervention@8**. It has resolved the issue. Let me know if I still need to add **OnTimeout: resume** in **ManualValidation@**.

Does it wait still? Whats the maximum wait time?

**Comment by hirendra-singh** [comment]
> > Hi @robe070, I have used **ManualValidation@1** task instead of **ManualIntervention@8**. It has resolved the issue. Let me know if I still need to add **OnTimeout: resume** in **ManualValidation@**.
> 
> Does it wait still? Whats the maximum wait time?

Yes, It wait and give the option to approve, I have added timeoutInMinutes: 43200 (30 days).

**Comment by robe070** [comment]
> Yes, It wait and give the option to approve, I have added timeoutInMinutes: 43200 (30 days).

Good. It needs to display the text and wait. Thats all that matters

**Comment by hirendra-singh** [comment]
> Why do you need a new set of variables - UpdatedGate? This is not the way to do it. You need to create a new version of `SetGateVariable.ps1` Call it `SetGateVariableTemplateCI.ps1` And set the Version in that script using TemplateCI

Removed the **UpdateGate** variable and created new **SetGateVariableTemplateCI.ps1**.

**Comment by robe070** [comment]
> Removed the **UpdateGate** variable and created new **SetGateVariableTemplateCI.ps1**.

You haven't checked in the script yet

**Comment by hirendra-singh** [comment]
> > Removed the **UpdateGate** variable and created new **SetGateVariableTemplateCI.ps1**.
> 
> You haven't checked in the script yet


Created new PR for SetGateVariableTemplateCI.ps1 file.
https://github.com/lansa/aws-templates/pull/34

**Comment by robe070** [comment]
Looking at [Pipelines - Run 20241009.16 (azure.com)](https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2587&view=results)

The region is not being set for each test. E.g. The Ireland stage is still being run in Sydney.

And I notice that in that pipeline there is one failure, but I can't locate the Cloud Formation stack to work out the error. Do you understand this failure?

**Comment by hirendra-singh** [comment]
> Looking at [Pipelines - Run 20241009.16 (azure.com)](https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2587&view=results)
> 
> The region is not being set for each test. E.g. The Ireland stage is still being run in Sydney.

Will check this out

> And I notice that in that pipeline there is one failure, but I can't locate the Cloud Formation stack to work out the error. Do you understand this failure?

Yes, It was typo error, missing char was there.

**Comment by robe070** [comment]
Please add link to pipeline run that shows all stacks created in correct regions and all stages successful. Then I can merge the PR.

**Comment by hirendra-singh** [comment]
> Please add link to pipeline run that shows all stacks created in correct regions and all stages successful. Then I can merge the PR.

Getting below error while change region from **ap-southeast-2** to other

**1. For us-east-1 region**

**Error in aws stack**

Parameter validation failed: parameter value subnet-08358c3721c83aadb for parameter name ELBSubnetIds does not exist, parameter value subnet-0c48beacbc5772e71 for parameter name ELBSubnetIds does not exist, parameter value ap-southeast-2a for parameter name AvailabilityZones does not exist, parameter value ap-southeast-2b for parameter name AvailabilityZones does not exist, parameter value ap-southeast-2c for parameter name AvailabilityZones does not exist, parameter value vpc-06adef7ecab120405 for parameter name CurrentVPC does not exist

https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2623&view=logs&j=8d0e5ca0-6830-562f-ba06-784a4c95c55c&t=3d34552a-c6e4-5790-6b4b-a7c933789fe9&l=55

**2. For other Region**
Stack creation request failed with error: 'S3 error: The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.

https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2633&view=logs&j=cf6161eb-4b30-5d40-e39a-265244057fda&t=cb71f675-4970-58c7-1699-5415f8e7e747&l=221

**Comment by robe070** [comment]
> > Please add link to pipeline run that shows all stacks created in correct regions and all stages successful. Then I can merge the PR.
> 
> Getting below error while change region from **ap-southeast-2** to other
> 
> **1. For us-east-1 region**
> 
> **Error in aws stack**
> 
> Parameter validation failed: parameter value subnet-08358c3721c83aadb for parameter name ELBSubnetIds does not exist, parameter value subnet-0c48beacbc5772e71 for parameter name ELBSubnetIds does not exist, parameter value ap-southeast-2a for parameter name AvailabilityZones does not exist, parameter value ap-southeast-2b for parameter name AvailabilityZones does not exist, parameter value ap-southeast-2c for parameter name AvailabilityZones does not exist, parameter value vpc-06adef7ecab120405 for parameter name CurrentVPC does not exist

Thats because you haven't passed in the appropriate VPC details. Look at the classic pipeline.

e.g. Paris CFN Deployment VPC params:
```
{
    "ParameterKey": "CurrentVPC",
    "ParameterValue": "$(CurrentVPC-eu-west-3)"
}, {
    "ParameterKey": "AvailabilityZones",
    "ParameterValue": "$(AvailabilityZones-eu-west-3)"
}, {
    "ParameterKey": "NumberOfAZs",
    "ParameterValue": "$(NumberOfAZs-eu-west-3)"
}, {
    "ParameterKey": "CreateVPC",
    "ParameterValue": "Yes"
}, {
    "ParameterKey": "ELBSubnetIds",
    "ParameterValue": "$(ELBSubnetIds-eu-west-3)"
}
```

**Review by robe070 — CHANGES_REQUESTED** [review]
Why are you not tagging these resources? Everything needs to be tagged.

**Review by robe070 — CHANGES_REQUESTED** [review]
Why have you removed the tags?

File: AWS Publish Templates.yml [hunk 1]
```diff
@@ -339,6 +347,10 @@ stages:
           targetFolder: '$(FolderNameProduction)'
           filesAcl: 'public-read'
 
+      - pwsh: |
+          Get-ChildItem '$(Pipeline.Workspace)/_lansa_aws-templates-mp/CloudFormationWindows'
+          Get-Content '$(Pipeline.Workspace)/_lansa_aws-templates-mp/CloudFormationWindows/lansa-stack-type-win.cfn.template'
+
```
Comment (robe070) [inline AWS Publish Templates.yml:353]: Please explain this change

Comment (hirendra-singh) [reply AWS Publish Templates.yml:353]: Both lines need to be commented/delete. It was added just to check whether s3 getting correct updated file or not.

**Pull request merged**